### DISCUSSION TITLE
Please comment on CCTZ bundling

### DISCUSSION THREAD

**Discussion by musicinmybrain** [abseil/abseil-cpp#1415] [General]
I see that `absl/time/internal/cctz/` contains a partial copy of https://github.com/google/cctz, forked with Abseil-specific changes.

As part of the [guidelines for bundling and duplication of system libraries](https://docs.fedoraproject.org/en-US/packaging-guidelines/#bundling) in Fedora Linux, I am required to publicly inquire about a path to supporting an external/system copy of this library.

Is this a future possibility, or will forked-and-bundled CCTZ sources always be required? That’s not a problem, but an opinion from an Abseil maintainer on the matter would be appreciated.

**Comment by derekmauro** [comment]
If you look at the commit history, at one point we did do just that: https://github.com/abseil/abseil-cpp/commit/af7882601aad93ada881486eeaabc562f1733961

In 2018 the decision was made to bundle the library. I remember being skeptical of the decision at the time. The reasoning was that at Google we really don't use CCTZ directly, we always use it through the Abseil library using Abseil types. It was thought that enough of the functionality of CCTZ would eventually make it into the C++ standard library so that CCTZ could be deprecated.

Anyway, we have no plans to change this decision, but we reserve the right to change our minds.

**Comment by musicinmybrain** [comment]
Thank you for your response! That’s exactly what I needed.
